src/move_actions.c

```diff
@@ -2644,7 +2644,7 @@ void sub_805A7D4(Entity * pokemon, Entity * target, Item *item, Position *pos)
   stackEntity.unk24 = 0;
   stackEntity.isVisible = TRUE;
   stackEntity.unk22 = 0;
-  stackEntity.info = (EntityInfo*) item;
+  stackEntity.axObj.info = (EntityInfo*) item; // This seems wrong!
```

Comment:
The `Item` fields line up with `EntityInfo`,, so this seems to work okay. It could be that the entity info is some kind of union, since it needs to represent either Pokémon, items, or traps.

Reply:
Yup, you can see it in `src/dungeon_util.c` where each of the info are casted to the respective types.